src/extractor/edge_based_graph_factory.cpp

```diff
@@ -431,8 +431,9 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
                     distance += profile_properties.traffic_signal_penalty;
                 }
 
+                const auto turn_angle = turn.angle > 180 ? 180 - turn.angle : turn.angle;
                 const int32_t turn_penalty =
-                    scripting_environment.GetTurnPenalty(180. - turn.angle);
+                    scripting_environment.GetTurnPenalty(turn_angle);
```

Comment:
might I ask where this is coming from? As you can see in https://github.com/Project-OSRM/osrm-backend/issues/2918, I have made tests using exactly `180-turn.angle` and you get just the results we want. 

From my tests an angle of 180 (resulting in 0 as call argument) is getting a turn penalty of zero.
From your new call value, we would get 7.4 seconds for going straight.
Osrm assumes 180 as angle for going straight. The turn function in the profile expects 0 for going straight.
The angle change is broken here.